Release 2.4.7 (#7165)
nautobot/nautobot: nautobot/docs/release-notes/version-2.4.md

<!-- towncrier release notes start -->

## v2.4.7 (2025-04-14)

### Added in v2.4.7

- [#4171](https://github.com/nautobot/nautobot/issues/4171) - Added `TYPE_NOTES` and `TYPE_URL` to SecretsGroupSecretTypeChoices.

> - [#6998](https://github.com/nautobot/nautobot/issues/6998) - Added browser and backend caching for `/api/swagger` openapi endpoint to speed up Swagger and Redoc loading time.

Reviewer: ```suggestion
- [#6998](https://github.com/nautobot/nautobot/issues/6998) - Added browser and backend caching for `/api/swagger` OpenAPI endpoint to speed up Swagger and Redoc loading time.
```

- [#7115](https://github.com/nautobot/nautobot/issues/7115) - Added bulk-edit support for the Relationship model.

> - [#7127](https://github.com/nautobot/nautobot/issues/7127) - Added Bulk Edit functionality for the ExportTemplate model.

Reviewer: ```suggestion
- [#7127](https://github.com/nautobot/nautobot/issues/7127) - Added bulk-edit support for the ExportTemplate model.
```

### Changed in v2.4.7

- [#6753](https://github.com/nautobot/nautobot/issues/6753) - Removed indentation of child locations on location details page.
- [#7114](https://github.com/nautobot/nautobot/issues/7114) - Changed "Locations" column in VLAN table to show `location.name` instead of `location.display` to avoid verbose location hierarchy.

### Fixed in v2.4.7

- [#5287](https://github.com/nautobot/nautobot/issues/5287) - Fixed text to be selectable on homepage panels.
- [#6923](https://github.com/nautobot/nautobot/issues/6923) - Fixed auto populating position field when creating Module Bay from Device Type details and bulk creating from Devices or Modules list.
- [#7101](https://github.com/nautobot/nautobot/issues/7101) - Fixed saving a new rack elevation view.
- [#7108](https://github.com/nautobot/nautobot/issues/7108) - Fixed various cases where CustomField-related `provision_field` and `delete_custom_field_data` background tasks were unnecessarily triggered.
- [#7108](https://github.com/nautobot/nautobot/issues/7108) - Added missing signal handler to remove custom field data from affected objects when `CustomField.content_types.clear()` is called.

> - [#7152](https://github.com/nautobot/nautobot/issues/7152) - Location.display now honors LOCATION_NAME_AS_NATURAL_KEY - meaning the display of location names will not render the full hierarchy, in places where only the name is relevant.

Reviewer: ```suggestion
- [#7152](https://github.com/nautobot/nautobot/issues/7152) - Changed `Location.display` to honor `LOCATION_NAME_AS_NATURAL_KEY` - meaning the display of location names will not render the full hierarchy, in places where only the name is relevant.
```

### Housekeeping in v2.4.7

> - [#6923](https://github.com/nautobot/nautobot/issues/6923) - `forms.js` refactor of `initializeSlugField` to use common logic with auto populate field like Module Bay position.

Reviewer: ```suggestion
- [#6923](https://github.com/nautobot/nautobot/issues/6923) - Refactored `initializeSlugField` to use common logic with other auto populated field like Module Bay position.
```
